sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_floating_point.asciidoc

```diff
@@ -0,0 +1,1160 @@
+|`OpArbitraryFloatCosINTEL`               | 5869
+|`OpArbitraryFloatSinCosINTEL`            | 5870
+|`OpArbitraryFloatSinPiINTEL`             | 5871
+|`OpArbitraryFloatCosPiINTEL`             | 5872
+|`OpArbitraryFloatSinCosPiINTEL`          | 5840
+|`OpArbitraryFloatASinINTEL`              | 5873
+|`OpArbitraryFloatASinPiINTEL`            | 5874
+|`OpArbitraryFloatACosINTEL`              | 5875
+|`OpArbitraryFloatACosPiINTEL`            | 5876
+|`OpArbitraryFloatATanINTEL`              | 5877
+|`OpArbitraryFloatATanPiINTEL`            | 5878
+|`OpArbitraryFloatATan2INTEL`             | 5879
+|`OpArbitraryFloatPowINTEL`               | 5880
+|`OpArbitraryFloatPowRINTEL`              | 5881
+|`OpArbitraryFloatPowNINTEL`              | 5882
+|`OpArbitraryFloatCastINTEL`              | 5841
+|`OpArbitraryFloatCastFromIntINTEL`       | 5842
+|`OpArbitraryFloatCastToIntINTEL`         | 5843
+|====
+
+== Modifications to the SPIR-V Specification Version 1.5
+
+After Section 3.16, add a new section "3.16a Subnormal Support" as follows:
+
+=== Subnormal Support
+
+Control whether subnormal support is enabled or not.
+
+[options="header"]
+[width="60%"]
+[cols="15%,85%"]
+|====
+| Value ^| Subnormal Support (`OpTypeInt`)
+| 0 | Flush subnormals numbers to zero on inputs and outputs
+| 1 | Enable support for operating on subnormal numbers
+|====
+
+After Section 3.16, add a new section "3.16d Rounding Accuracy" as follows:
+
+=== Rounding Accuracy
+
+Controls whether rounding operations can be relaxed to trade correctness for improved resource utilization.
+
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode     ^| Behavior
+| 0      | CORRECT   | Conform to the IEEE-754 standard for rounding accuracy
+| 1      | FAITHFUL  | Allow some tolerance for error (within 1ULP of the IEEE-754 standard) for rounding. +
+The returned result is one of the two floating point values closest to the mathematical result.
+
+This mode is useful for devices that can trade CORRECT rounding for improved resource utilization.
+|====
+
+=== Capability
+Modify Section 3.31, *Capability*, adding a row to the Capability table:
+
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 5845 | *ArbitraryPrecisionFloatingPointINTEL* +
+
+Allows the use of various operations for arbitrary precision floating-point math
+
+| *ArbitraryPrecisionIntegersINTEL*, *Kernel* |
+|====
+
+=== Instructions
+
+In Section 3.32.13, *Arithmetic Instructions*, add the following instructions:
+
+[cols="12", width="100%"]
+|=====
+11+<|*OpArbitraryFloatAddINTEL* +
+
+Two `OpTypeInts` representing two arbitrary precision floating point numbers (ihc_float) are passed in as _A_ and _B_ and the result of `A+B` is returned in _Result_.
+
+_Result Type_ is an `OpTypeInt` of width `1+Eout+Mout` and is the type of _Result_.
+
+_M*_ contains the width of the mantissa of the floating point types within _Result_, _A_, and _B_.
```

Comment:
Personally, I would prefer to have clear mapping between literal and argument which it corresponds to, but I guess it could be inferred with current wording as well: `Mout` for result, `M1` for the first argument and `M2` for the second

Probably would be good to hear more feedback from someone else

Reply:
@AlexeySachkov any suggestions on what you would rename these to?